pkgs/tools/filesystems/simg2img/default.nix

```diff
@@ -13,13 +13,17 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ zlib ];
 
+  preConfigure = lib.optionals stdenv.isDarwin ''
+    substituteInPlace Makefile --replace "g++" "/usr/bin/g++"
+  '';
+
   makeFlags = [ "PREFIX=$(out)" ];
```

Comment:
Let's use the compiler from the `stdenv` by updating `makeFlags`:

```
makeFlags = [ "PREFIX=$(out)" "CXX:=$(CXX)" "DEP_CXX:=$(CXX)" ];
```

Reply:
Updating just the non-standard `DEP_CXX` works on both darwin and `clangStdenv` on linux.